From a299448603dd3487ef37470a316eca29411fca36 Mon Sep 17 00:00:00 2001
From: Petr Lautrbach <plautrba@redhat.com>
Date: Thu, 6 Oct 2016 10:45:09 +0200
Subject: [PATCH 43/43] sandbox: Always use seunshare

seunshare was changed in order not to require -t and -h options.

sandbox uses seunshare always now even when there's no mount points
specified by a user.

Resolves: CVE-2016-7545
---
 sandbox/sandbox     | 35 +++++++++++++++--------------------
 sandbox/seunshare.c |  2 ++
 2 files changed, 17 insertions(+), 20 deletions(-)

diff --git a/sandbox/sandbox b/sandbox/sandbox
index f4f6c29..869ff98 100644
--- a/sandbox/sandbox
+++ b/sandbox/sandbox
@@ -421,29 +421,24 @@ sandbox [-h] [-l level ] [-[X|M] [-H homedir] [-T tempdir]] [-I includefile ] [-
                   if self.__mount:
                          cmds +=  [ "-t", self.__tmpdir, "-h", self.__homedir ]
 
-                         if self.__options.X_ind:
-                                if self.__options.dpi:
-                                       dpi = self.__options.dpi
-                                else:
-                                       import gtk
-                                       dpi = str(gtk.settings_get_default().props.gtk_xft_dpi/1024)
-
-                                xmodmapfile = self.__homedir + "/.xmodmap"
-                                xd = open(xmodmapfile,"w")
-                                subprocess.Popen(["/usr/bin/xmodmap","-pke"],stdout=xd).wait()
-                                xd.close()
+                  if self.__options.X_ind:
+                         if self.__options.dpi:
+                                dpi = self.__options.dpi
+                         else:
+                                import gtk
+                                dpi = str(gtk.settings_get_default().props.gtk_xft_dpi/1024)
 
-                                self.__setup_sandboxrc(self.__options.wm)
+                         xmodmapfile = self.__homedir + "/.xmodmap"
+                         xd = open(xmodmapfile,"w")
+                         subprocess.Popen(["/usr/bin/xmodmap","-pke"],stdout=xd).wait()
+                         xd.close()
 
-                                cmds += [ "--", SANDBOXSH, self.__options.windowsize, dpi ]
-                         else:
-                                cmds += [ "--" ] + self.__paths
-                         return subprocess.Popen(cmds).wait()
+                         self.__setup_sandboxrc(self.__options.wm)
 
-                  selinux.setexeccon(self.__execcon)
-                  rc = subprocess.Popen(self.__cmds).wait()
-                  selinux.setexeccon(None)
-                  return rc
+                         cmds += [ "--", SANDBOXSH, self.__options.windowsize, dpi ]
+                  else:
+                         cmds += [ "--" ] + self.__paths
+                  return subprocess.Popen(cmds).wait()
 
            finally:
                   for i in self.__paths:
diff --git a/sandbox/seunshare.c b/sandbox/seunshare.c
index ed7f6e7..967d39f 100644
--- a/sandbox/seunshare.c
+++ b/sandbox/seunshare.c
@@ -679,10 +679,12 @@ int main(int argc, char **argv) {
 		}
 	}
 
+	/*
 	if (! homedir_s && ! tmpdir_s) {
 		fprintf(stderr, _("Error: tmpdir and/or homedir required\n %s\n"), USAGE_STRING);
 		return -1;
 	}
+	*/
 
 	if (argc - optind < 1) {
 		fprintf(stderr, _("Error: executable required\n %s\n"), USAGE_STRING);
-- 
1.8.3.1

